src/lib/ringbuffer.cpp

```diff
@@ -25,6 +25,11 @@
 
 #ifdef USE_MLOCK
 	#include <sys/mman.h>
+	#ifdef __HAIKU__
```

Comment:
Why do you include `<sys/mman.h>` for Haiku at all?

Also, can't you disable `USE_MLOCK` in `cmake/process_project.txt` instead? The way it is now, `cmake/process_project.txt` will print that mlock is being used on Haiku.

Reply:
good point. hope it suits better now.